source/frontend/EditorFactory.cpp

```diff
@@ -7,65 +7,95 @@
 #include <frontend/editors/BlightEditor.hpp>
 #include <frontend/editors/BlmapEditor.hpp>
 #include <frontend/editors/BtkEditor.hpp>
+#include <frontend/editors/RarcEditor.hpp>
 #include <frontend/legacy_editor/EditorWindow.hpp>
 #include <frontend/level_editor/LevelEditor.hpp>
 
 #include <frontend/file_host.hpp>
 #include <librii/szs/SZS.hpp>
+#include <librii/rarc/RARC.hpp>
+#include <librii/u8/U8.hpp>
 
 namespace riistudio::frontend {
 
 //! Create an editor from the file data specified. Returns nullptr on failure.
 std::unique_ptr<IWindow> MakeEditor(FileData& data) {
   rsl::info("Opening file: {}", data.mPath.c_str());
 
-  std::span<const u8> span(data.mData.get(), data.mData.get() + data.mLen);
+  std::span<u8> raw_span(data.mData.get(), data.mData.get() + data.mLen);
```

Comment:
Still not convinced of the `extended_span` approach, would probably prefer individual editors (RARC/U8) manage compression for now.